File: src/main/java/net/frozenblock/configurableeverything/datafixer/util/DataFixEntry.java
package net.frozenblock.configurableeverything.datafixer.util;

import com.mojang.serialization.Codec;
import com.mojang.serialization.codecs.RecordCodecBuilder;
import java.util.List;

public record DataFixEntry(String type, List<Fixer> fixers) {

	public static final Codec<DataFixEntry> CODEC = RecordCodecBuilder.create(instance ->
			instance.group(
					Codec.STRING.fieldOf("type").forGetter(DataFixEntry::type),
					Codec.list(Fixer.CODEC).fieldOf("fixers").forGetter(DataFixEntry::fixers)
			).apply(instance, DataFixEntry::new)
	);
}

File: src/main/java/net/frozenblock/configurableeverything/datafixer/util/DataFixEntry.kt
package net.frozenblock.configurableeverything.datafixer.util

import com.mojang.serialization.Codec
import com.mojang.serialization.codecs.RecordCodecBuilder
import net.frozenblock.configurableeverything.util.mutListOf

data class DataFixEntry(
    @JvmField var type: String,
    @JvmField var fixers: MutableList<Fixer>
) {
    companion object {
        @JvmField
        val CODEC: Codec<DataFixEntry> = RecordCodecBuilder.create { instance ->
            instance.group(
                Codec.STRING.fieldOf("type").forGetter(DataFixEntry::type),
                Fixer.CODEC.mutListOf().fieldOf("fixers").forGetter(DataFixEntry::fixers)
            ).apply(instance, ::DataFixEntry)
        }
    }
}
